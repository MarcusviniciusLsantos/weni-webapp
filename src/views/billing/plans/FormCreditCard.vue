<template>
  <div class="billing-add-credit-card">
    <UnnnicInput
      v-model="$store.state.BillingSteps.billing_details.cpfOrCnpj"
      :label="$t('billing.card.cpf_or_cnpj')"
      :type="cpfOrCnpjError ? 'error' : 'normal'"
      :message="cpfOrCnpjError ? $t(`errors.${cpfOrCnpjError}`) : null"
      @input="$emit('update:errors', { ...errors, cpfOrCnpj: '' })"
    />
    <UnnnicInput
      v-model="$store.state.BillingSteps.billing_details.name"
      :label="$t('billing.card.name')"
      :placeholder="$t('billing.card.name_placeholder')"
      :type="nameError ? 'error' : 'normal'"
      :message="nameError ? $t(`errors.${nameError}`) : null"
      @input="$emit('update:errors', { ...errors, name: '' })"
    />
    <div class="billing-add-credit-card__bottom">
      <div>
        <div class="label">
          <label for="card-number">
            {{ $t('billing.card.number') }}
          </label>
        </div>

        <div id="card-number"></div>
      </div>

      <div>
        <div class="label">
          <label for="card-expiry">
            {{ $t('billing.card.due_date') }}
          </label>
        </div>

        <div id="card-expiry"></div>
      </div>

      <div>
        <div class="label">
          <label for="card-cvc">{{ $t('billing.card.ccv') }}</label>
        </div>

        <div id="card-cvc"></div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'BillingModal',

  props: {
    flow: String,

    errors: Object,
  },

  data() {
    return {};
  },

  computed: {
    cpfOrCnpjError() {
      return this.errors?.cpfOrCnpj;
    },

    nameError() {
      return this.errors?.name;
    },
  },
};
</script>

<style lang="scss" scoped>


.billing-add-credit-card {
  .label {
    font-weight: 400;
    line-height: 1.375rem;
    font-size: 0.875rem;
    color: #67738b;
    margin-bottom: 0.5rem;
  }

  > .unnnic-form {
    margin-bottom: $unnnic-spacing-stack-md;
  }

  &__bottom {
    display: flex;
    > div:first-child {
      width: 60%;
    }
    > div:nth-child(2) {
      width: 37%;
    }
    > div:last-child {
      width: 18%;
    }

    > div {
      margin: 0 $unnnic-inline-sm $unnnic-spacing-stack-md 0;
      &:first-child {
        margin-left: 0;
      }
      &:last-child {
        margin-right: 0;
      }
    }
  }
}
</style>
