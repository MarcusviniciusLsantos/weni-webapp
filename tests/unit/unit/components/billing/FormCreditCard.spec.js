import { shallowMount, createLocalVue } from '@vue/test-utils';
import FormCreditCard from '@/views/billing/plans/FormCreditCard.vue';
import i18n from '@/utils/plugins/i18n';

import Vuex from 'vuex';
import { org } from '../../../__mocks__';
const localVue = createLocalVue();
localVue.use(Vuex);

describe('FormCreditCard.vue', () => {
  let wrapper;

  let actions;
  let store;
  let getters;
  let state;

  beforeEach(() => {
    actions = {};
    state = {
      BillingSteps: {
        billing_details: {
          address: {
            country: 'BR',
            state: 'AL',
          },
        },
      },
    };
    getters = {
      currentOrg() {
        return org;
      },
    };
    store = new Vuex.Store({
      actions,
      getters,
      state,
    });
    wrapper = shallowMount(FormCreditCard, {
      localVue,
      i18n,
      propsData: {
        flow: 'test',
      },
      store,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        unnnicToolTip: true,
        unnnicButton: true,
        unnnicSelect: true,
        unnnicInput: true,
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
