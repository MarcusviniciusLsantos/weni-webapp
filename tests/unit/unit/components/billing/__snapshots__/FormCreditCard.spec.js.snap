// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormCreditCard.vue > renders a snapshot 1`] = `
<div
  class="billing-add-credit-card"
  data-v-0a8f8dba=""
>
  <unnnicinput
    data-v-0a8f8dba=""
    label="some specific text"
    type="normal"
  />
  <unnnicinput
    data-v-0a8f8dba=""
    label="some specific text"
    placeholder="some specific text"
    type="normal"
  />
  <div
    class="billing-add-credit-card__bottom"
    data-v-0a8f8dba=""
  >
    <div
      data-v-0a8f8dba=""
    >
      <div
        class="label"
        data-v-0a8f8dba=""
      >
        <label
          data-v-0a8f8dba=""
          for="card-number"
        >
           some specific text 
        </label>
      </div>
      <div
        data-v-0a8f8dba=""
        id="card-number"
      />
    </div>
    <div
      data-v-0a8f8dba=""
    >
      <div
        class="label"
        data-v-0a8f8dba=""
      >
        <label
          data-v-0a8f8dba=""
          for="card-expiry"
        >
           some specific text 
        </label>
      </div>
      <div
        data-v-0a8f8dba=""
        id="card-expiry"
      />
    </div>
    <div
      data-v-0a8f8dba=""
    >
      <div
        class="label"
        data-v-0a8f8dba=""
      >
        <label
          data-v-0a8f8dba=""
          for="card-cvc"
        >
          some specific text
        </label>
      </div>
      <div
        data-v-0a8f8dba=""
        id="card-cvc"
      />
    </div>
  </div>
</div>
`;
